ui/packages/consul-ui/ember-cli-build.js

```diff
@@ -66,6 +68,30 @@ module.exports = function(defaults, $ = process.env) {
       'templates/debug.hbs',
       'components/debug/**/*.*'
     ])
+    // inspect *-debug configuration files for files to exclude
+    excludeFiles = apps.reduce(
+      (prev, item) => {
+        return ['services', 'routes'].reduce(
+          (prev, type) => {
+            const path = `${item.path}/vendor/${item.name}/${type}-debug.js`;
+            if(exists(path)) {
+              return Object.entries(JSON.parse(require(path)[type])).reduce(
+                (prev, [key, definition]) => {
+                  if(typeof definition.class !== 'undefined') {
+                    return prev.concat(`${definition.class.replace(`${item.name}/`, '')}.js`);
+                  }
+                  return prev;
+                },
+                prev
+              );
+            }
+            return prev;
+          },
+          prev
+        )
+      },
+      excludeFiles
+    );
```

Comment:
This is one of my favourite parts. I've eternally found it difficult in ember to be specific about the files I do/don't want complied into my final build. For example, I might want certain helpers only to be included during development, maybe I want to inject a mocked ember Service in place of the real ember Service only during development.

It's non-obvious how to keep these development only files out of a production build in Ember.

Previously (and still currently after this PR) we use wildcard/globbing to say "anything that ends in `-debug.js` please don't compile in for production" i.e. a convention (you'll see this if you explan the code above line 68).

In this PR we reuse the configuration that we already use to do the actual injection of the dev-time-only Services/Helpers to figure out exactly what files need excluding i.e. using the explicit configuration over the convention.

Currently we still use the `*-debug.js` convention also, just as I want to make sure we can use the same approach for all-the-things including templates etc. Once we get there we will eventually remove the `*-debug.js` convention.